src/librustc/lint/builtin.rs

```diff
@@ -1628,6 +1628,52 @@ impl LintPass for MissingCopyImplementations {
 }
 
 declare_lint! {
+    pub SHADOWING_MACROS,
+    Warn,
```

Comment:
Due to this being a known *feature* of the macro system, this should be `Allow` by default.